### DISCUSSION TITLE
CHP Cost and Performance Defaults Updates

### DISCUSSION THREAD

**Discussion by Bill-Becker** [NREL/REopt-API-Analysis#122] [Announcements]
REopt has updated cost and performance default values for CHP prime movers 1) Reciprocating Engine, 2) Microturbine and 3) Combustion Turbine based on data from updated DOE CHP Fact Sheets (to be published soon).

See Section 14.8 of the [REopt User Manual](https://reopt.nrel.gov/tool/reopt-user-manual.pdf) for the update cost and performance parameters by size class for these prime movers which are supported from the data of the updated fact sheets in Appendix A

**Comment by JackBoDree** [comment]
I think I accepted the default figures in all other required field.  Do I need to retype the default figures for the system to register?

I used a CHP/grid case with hot water.

For emergency power, I used 24 hours and kept the standard outages.

Jacques

Sent from my iPhone

On Oct 30, 2023, at 19:02, Bill Becker ***@***.***> wrote:

﻿
You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

Could you let me know all the inputs you put in, besides the large hotel building type and the fuel cost of $9/MMBtu?

**Reply by Bill-Becker** [reply]
location? Can you please reply to this post instead of creating a separate post? I'm going to remove some of these initial posts to avoid taking up too much space in this forum thread.

**Reply by JackBoDree** [reply]
Location was: Martinsville,VA and I used Appalachian Time of Day electricity pricing.

**Reply by Bill-Becker** [reply]
Ah, I figured it out. That error message is unclear, but it's an infeasible problem. By default with CHP, we set the max size REopt can choose equal to 2x the CHP size based on the average heating load. In this case, that size is not large enough to serve the critical electric load during the outage, which the model has set as a constraint. The solution is to increase the maximum capacity to something like the peak/max electric load.

Here:
![image](https://github.com/NREL/REopt-Analysis-Scripts/assets/42586683/aa4bd335-1bce-43ff-84c7-965d5461a446)

**Reply by Bill-Becker** [reply]
I was overconfident, increasing the maximum capacity is not fixing the problem. I am still investigating.

**Reply by Bill-Becker** [reply]
From @JackBoDree:
Can I simply remove the resiliency requirement for now?  Maybe it will  run?
Setting the electricity price as a single custom number  (as opposed to the dropdown menu driven options) at least helped me run the simole configurations.

Other surprise is that no matter which electricity or gas price I use, there is ni savings with CHP.

**Reply by Bill-Becker** [reply]
It's definitely a CHP+Resilience issue, so if you take one of those out of the scenario it will run.

**Reply by JackBoDree** [reply]
Ok, will try.

**Reply by Bill-Becker** [reply]
I found the bug!! This will be deployed -> fixed tomorrow/Thursday.

**Reply by Bill-Becker** [reply]
In case you've tried to test this again, I fixed the root cause but then discovered another. The timing of the outages overlaps with the unavailability profile of CHP, so there were some hours where CHP was not able to serve the load, causing another infeasibility. I'm working on fixing that, but this scenario will run if you move the culprit unavailability period or the outage. I'm going to ignore unavailability during the outage time so this issue doesn't come up.

**Reply by Bill-Becker** [reply]
The conflicting maintenance schedule and outages issue has been fixed as well now.

**Comment by JackBoDree** [comment]
Thank you!
